src/extra/imgui.hpp

```diff
@@ -35,5 +37,85 @@ struct Context {
   //   context = ::ImGui::CreateContext();
   // }
 };
+
+struct Enums {
+#define REGISTER_ENUM(_NAME_, _CC_)                                            \
+  static constexpr uint32_t _NAME_##CC = _CC_;                                 \
+  static inline EnumInfo<_NAME_> _NAME_##EnumInfo{#_NAME_, CoreCC,             \
+                                                  _NAME_##CC};                 \
+  static inline Type _NAME_##Type = Type::Enum(CoreCC, _NAME_##CC)
+
+#define REGISTER_FLAGS(_NAME_, _CC_)                                           \
+  static constexpr uint32_t _NAME_##CC = _CC_;                                 \
+  static inline FlagsInfo<_NAME_> _NAME_##EnumInfo{#_NAME_, CoreCC,            \
+                                                   _NAME_##CC};                \
+  static inline Type _NAME_##Type = Type::Enum(CoreCC, _NAME_##CC)
+
+#define REGISTER_FLAGS_EX(_NAME_, _CC_)                                        \
+  REGISTER_FLAGS(_NAME_, _CC_);                                                \
+  static inline Type _NAME_##SeqType = Type::SeqOf(_NAME_##Type);              \
+  static inline Type _NAME_##VarType = Type::VariableOf(_NAME_##Type);         \
+  static inline Type _NAME_##VarSeqType = Type::VariableOf(_NAME_##SeqType)
```

Comment:
These macros could be moved to a more general header file, but I'm not sure which one.

`REGISTER_CBLOCK` is in `include\blockwrapper.hpp` but it doesn't seem like the right place for them.

Reply:
Generally Utility or Chainblocks.hpp

Reply:
Actually, neither can work here since we need `EnumInfo` to be defined. I will put it in `foundation.hpp`.